src/Harald.WebApi/Application/EventHandlers/SlackMemberJoinedCapabilityDomainEventHandler.cs

```diff
@@ -29,48 +29,53 @@ public class SlackMemberJoinedCapabilityDomainEventHandler : IEventHandler<Membe
 
         public async Task HandleAsync(MemberJoinedCapabilityDomainEvent domainEvent)
         {
-            var capability = await _capabilityRepository.Get(domainEvent.Payload.CapabilityId);
-
-            if (capability == null)
+            var capabilities = await _capabilityRepository.GetById(domainEvent.Payload.CapabilityId);
+            
+            // TODO not cool to do a foreach over so much code
```

Comment:
Yeah, plus is this safe in case of partials errors?

Reply:
They have they same logic as before the foreach. I agree with your concern.